### ISSUE TITLE
Kubeapps Helm Chart does not allow configuring NGINX for large headers

### ISSUE THREAD

**Issue by plnordquist** [kubeapps/kubeapps#1571]
### Description:

With the current configuration of the kubeapps Helm chart, I am unable to configure the nginx serving the dashboard and the frontend nginx to support large headers with oauth2-proxy.

### Steps to reproduce the issue:

1. Configure an OIDC provider for Kubernetes, in my case Keycloak
1. Deploy the kubeapps helm chart with oauth2-proxy configured to talk to Keycloak
2. Attempt to load kubeapps in the browser

### Describe the results you received:

NGINX returns a 400 Bad Request Request Header Or Cookie Too Large

### Describe the results you expected:

The application loads properly.

### Additional information you deem important (e.g. issue happens only occasionally):

My Keycloak is configured to add group membership to the token and that's how the tokens are so large.  The workaround I currently use is to add `large_client_header_buffers 4 32k;` to the vhost.conf file inside the configmaps that control nginx.  For the `kubeapps-frontend-config` configmap I've placed the directive above the server block in the vhost.conf file.  For the `kubeapps-internal-dashboard-config` configmap I've placed the directive above inside the server block above the location block in the vhost.conf file.  I'm also running the oauth2-proxy outside of the Helm chart and I've customized the deployment as referenced here https://github.com/kubeapps/kubeapps/issues/1557#issuecomment-596741738.  I can try deploying the chart with oauth2-proxy inside the chart if it's necessary to test this issue fully.

### Version of Helm, Kubeapps and Kubernetes:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.1.1", GitCommit:"afe70585407b420d0097d07b21c47dc511525ac8", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

- Output of `helm list <kubeapps-release-name>`:

```
NAME                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
kubeapps                services        14              2020-03-06 11:48:36.334056678 -0800 PST deployed        kubeapps-3.4.3                  v1.8.2
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"be3d344ed06bff7a4fc60656200a93c74f31f9a4", GitTreeState:"clean", BuildDate:"2020-02-11T19:34:02Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"be3d344ed06bff7a4fc60656200a93c74f31f9a4", GitTreeState:"clean", BuildDate:"2020-02-11T19:24:46Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by absoludity** [comment]
Thanks @plnordquist . Yes, it looks like we should be enabling a chart option for custom settings for both vhost files (or at least, an option to support large headers).

>  For the `kubeapps-frontend-config` configmap I've placed the directive above the server block in the vhost.conf file. For the `kubeapps-internal-dashboard-config` configmap I've placed the directive above inside the server block above the location block in the vhost.conf file.

Did you try including them in the same place for both? (ie. before the server block for both, or just inside the server block before the first location for both). That way we could just add a template option in the same place, depending on what could potentially be required by added settings.

**Comment by plnordquist** [comment]
Adding the `large_client_header_buffer` directive to the server block just before the first location section worked just fine.

**Comment by andresmgot** [comment]
This was implemented in #2027
